### PR TITLE
Add extra features/checking required for operation mango to function

### PR DESCRIPTION
**Pull request by WGibbs15**

### CODE REVIEW

File: angr/analyses/reaching_definitions/dep_graph.py [hunk 1]
```diff
@@ -145,12 +145,21 @@ def add_dependencies_for_concrete_pointers_of(self,
             )
         ))
 
+        #concretize ddresses where possible
```
Comment (ltfish) [inline angr/analyses/reaching_definitions/dep_graph.py:148]: 1. `ddresses` -> `addresses`
2. Add a space after `#`.

File: angr/analyses/reaching_definitions/dep_graph.py [hunk 2]
```diff
@@ -145,12 +145,21 @@ def add_dependencies_for_concrete_pointers_of(self,
...
+        concrete_known_pred_addresses = []
+        for address in known_predecessor_addresses:
+            if isinstance(address, claripy.ast.Base):
+                if address.concrete:
+                    concrete_known_pred_addresses.append(address._model_concrete.value)
+            else:
```
Comment (ltfish) [inline angr/analyses/reaching_definitions/dep_graph.py:154]: What's the type of `address` under `else`?

**Review by ltfish — CHANGES_REQUESTED** [review]

File: angr/analyses/reaching_definitions/dep_graph.py (hunk 2 above)
Comment (WGibbs15) [reply angr/analyses/reaching_definitions/dep_graph.py:154]: `int` type addresses

Comment (WGibbs15) [reply angr/analyses/reaching_definitions/dep_graph.py:154]: This line -> https://github.com/angr/angr/blob/79f484e78aca77bcc00f79374dbb7a74ac0bdcc9/angr/analyses/reaching_definitions/dep_graph.py#L161
would throw an error if there were any `claripy.ast.Base` objects in the list.
The above loop just makes sure that everything is safely comparable.

**Review by ltfish — APPROVED** [review]

**Comment by ltfish** [comment]
Why is the CI failing?

**Pull request merged**